.github/workflows/github_pages.yml

```diff
@@ -0,0 +1,61 @@
+name: Deploy to GitHub Pages
+on:
+  workflow_call:
+    inputs:
+      settings:
+        required: true
+        description: "The path to your Pelican settings file (`pelican`'s `--settings` option), for example: 'publishconf.py'"
+        type: string
+      requirements:
+        required: false
+        default: "pelican"
+        description: "The Python requirements to install, for example to enable markdown and typogrify use: 'pelican[markdown] typogrify'"
+        type: string
```

Comment:
I think there is an undocumented feature of writing `"-r requirements.txt"` here given that repo contains that file, right? Might be worth mentioning :).

Reply:
I added a commit for this.